File: funciones/conceptos_compras.cs
using System;
using System.Linq;
using System.Windows.Forms;

namespace Centrex.Funciones
{

    static class conceptos_compra
    {

        // ************************************ FUNCIONES DE CONCEPTOS DE COMPRA **********************
        public static ConceptoCompraEntity info_concepto_compra(int id_concepto)
        {
                        try
            {
                using (CentrexDbContext context = new CentrexDbContext())
                {
                    var conceptoCompraEntity = context.ConceptoCompraEntity.FirstOrDefault(c => c.IdConceptoCompra == id_concepto);                    

                    if (conceptoCompraEntity is not null)
                    {                        
                        return conceptoCompraEntity;
                    }
                    else
                    {
                        return null;
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error al obtener concepto de compra:" + ex.Message, "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return null;
            }
        }

        public static bool addConcepto_compra(ConceptoCompraEntity Concepto)
        {
            try
            {
                using (CentrexDbContext context = new CentrexDbContext())
                {
                    var conceptoEntity = new ConceptoCompraEntity()
                    {
                        Concepto = Concepto.Concepto,
                        Activo = Concepto.Activo
                    };

                    context.ConceptoCompraEntity.Add(conceptoEntity);
                    context.SaveChanges();
                    return true;
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
                return false;
            }
        }

        public static bool updateConcepto_compra(ConceptoCompraEntity Concepto, bool borra = false)
        {
            try
            {
                using (CentrexDbContext context = new CentrexDbContext())
                {
                    var conceptoEntity = context.ConceptoCompraEntity.FirstOrDefault(c => c.IdConceptoCompra == Concepto.IdConceptoCompra);

                    if (conceptoEntity is not null)
                    {
                        if (borra == true)
                        {
                            conceptoEntity.Activo = false;
                        }
                        else
                        {
                            conceptoEntity.Concepto = Concepto.Concepto;
                            conceptoEntity.Activo = Concepto.Activo;
                        }

                        context.SaveChanges();
                        return true;
                    }
                    else
                    {
                        return false;
                    }
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
                return false;
            }
        }

        public static bool borrarConcepto_compra(ConceptoCompraEntity Concepto)
        {
            try
            {
                using (CentrexDbContext context = new())
                {
                    var conceptoEntity = context.ConceptoCompraEntity.FirstOrDefault(c => c.IdConceptoCompra == Concepto.IdConceptoCompra);

                    if (conceptoEntity is not null)
                    {


                        context.ConceptoCompraEntity.Remove(conceptoEntity);
                        context.SaveChanges();
                        return true;
                    }
                    else
                    {
                        return false;
                    }
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message.ToString());
                return false;
            }
        }
        // ************************************ FUNCIONES DE CONCEPTOS COMPRA **********************
    }
}

File: __Viejo/funciones/conceptos_compras.vb

Imports System.Data.SqlClient
Module conceptos_compra

    ' ************************************ FUNCIONES DE CONCEPTOS DE COMPRA **********************
    Public Function info_concepto_compra(ByVal id_concepto As String) As concepto_compra
        Dim tmp As New concepto_compra
        Dim sqlstr As String

        Try
            sqlstr = "SELECT id_concepto_compra, concepto, activo " +
                    "FROM conceptos_compra " +
                    "WHERE id_concepto_compra = '" + id_concepto.ToString + "'"

            'Crea y abre una nueva conexión
            abrirdb(serversql, basedb, usuariodb, passdb)

            'Propiedades del SqlCommand
            Dim comando As New SqlCommand
            With comando
                .CommandType = CommandType.Text
                .CommandText = sqlstr
                .Connection = CN
            End With

            Dim da As New SqlDataAdapter 'Crear nuevo SqlDataAdapter
            Dim dataset As New DataSet 'Crear nuevo dataset

            da.SelectCommand = comando

            'llenar el dataset
            da.Fill(dataset, "Tabla")
            tmp.id_concepto_compra = dataset.Tables("tabla").Rows(0).Item(0).ToString
            tmp.concepto = dataset.Tables("tabla").Rows(0).Item(1).ToString
            tmp.activo = dataset.Tables("tabla").Rows(0).Item(2).ToString
            cerrardb()
            Return tmp
        Catch ex As Exception
            'MsgBox(ex.Message.ToString)
            tmp.concepto = "error"
            cerrardb()
            Return tmp
        End Try
    End Function

    Public Function addConcepto_compra(ByVal concepto As concepto_compra) As Boolean
        abrirdb(serversql, basedb, usuariodb, passdb)

        Dim sqlstr As String
        Dim mytrans As SqlTransaction
        Dim Comando As New SqlCommand

        mytrans = CN.BeginTransaction()

        Try
            sqlstr = "INSERT INTO conceptos_compra (concepto, activo) VALUES ('" +
                        concepto.concepto + "', '" + concepto.activo.ToString + "')"
            Comando = New SqlCommand(sqlstr, CN) With {.Transaction = mytrans}
            Comando.ExecuteNonQuery()

            mytrans.Commit()
            cerrardb()
            Return True
        Catch ex As Exception
            MsgBox(ex.Message)
            cerrardb()
            Return False
        End Try
    End Function

    Public Function updateConcepto_compra(ByVal concepto As concepto_compra, Optional borra As Boolean = False) As Boolean
        abrirdb(serversql, basedb, usuariodb, passdb)

        Dim sqlstr As String
        Dim mytrans As SqlTransaction
        Dim Comando As New SqlCommand

        mytrans = CN.BeginTransaction()

        Try
            If borra = True Then
                sqlstr = "UPDATE conceptos_compra SET activo = '0' WHERE id_concepto_compra = '" + concepto.id_concepto_compra.ToString + "'"
            Else
                sqlstr = "UPDATE conceptos_compra SET concepto = '" + concepto.concepto + "', activo = '" + concepto.activo.ToString +
                            "' WHERE id_concepto_compra = '" + concepto.id_concepto_compra.ToString + "'"
            End If
            Comando = New SqlCommand(sqlstr, CN) With {.Transaction = mytrans}
            Comando.ExecuteNonQuery()

            mytrans.Commit()
            cerrardb()
            Return True
        Catch ex As Exception
            MsgBox(ex.Message)
            cerrardb()
            Return False
        End Try
    End Function

    Public Function borrarConcepto_compra(ByVal concepto As concepto_compra) As Boolean
        abrirdb(serversql, basedb, usuariodb, passdb)

        Dim sqlstr As String = ""
        Dim mytrans As SqlTransaction
        Dim Comando As New SqlCommand

        mytrans = CN.BeginTransaction()

        Try
            sqlstr = "DELETE FROM conceptos_compra WHERE id_concepto_compra = '" + concepto.id_concepto_compra.ToString + "'"
            Comando = New SqlCommand(sqlstr, CN) With {.Transaction = mytrans}
            Comando.ExecuteNonQuery()

            mytrans.Commit()
            cerrardb()
            Return True
        Catch ex As Exception
            MsgBox(ex.Message.ToString)
            cerrardb()
            Return False
        End Try
    End Function
    ' ************************************ FUNCIONES DE CONCEPTOS COMPRA **********************
End Module
